x/crosschain/keeper/evm_hooks.go

```diff
@@ -72,12 +73,25 @@ func (k Keeper) ProcessLogs(ctx sdk.Context, logs []*ethtypes.Log, emittingContr
 				return err
 			}
 		}
+		// We were able to parse the ZRC20 withdrawal event. However, we were unable to process it as the information was incorrect
+		// This means that there are some funds locked in the contract which cannot have an outbound , this can be a candidate for a refund
+		// TODO : Consider returning error or auto refunding the funds to the user
+		if err != nil && eventWithdrawal != nil {
```

Comment:
```suggestion
		} else if eventWithdrawal != nil {
		// We were able to parse the ZRC20 withdrawal event. However, we were unable to process it as the information was incorrect
		// This means that there are some funds locked in the contract which cannot have an outbound , this can be a candidate for a refund
		// TODO : Consider returning error or auto refunding the funds to the user
```
Would be clearer, `err != nil` is already true if `err == nil` doesn't check

Reply:
5e5e25dc7bdcf78eee2a5ea5fe09160251a0188e

Reply:
Can
```
if err := k.ProcessZRC20WithdrawalEvent(ctx, eventWithdrawal, emittingContract, txOrigin); err != nil {
	return err
}
```
modify `eventWithdrawal`? If not I still think we should use a `else` to make the flow of the program more clear

Reply:
aaea061766d4b155b32783fcb417ae9388328e95